protocols/499472/499472.ot2.py

```diff
@@ -0,0 +1,274 @@
+        tipvol -= 20
+        if tipvol == 0 and well > 6:
+            s50.drop_tip()
+
+    pick_up(s50)
+    for well in sampwells:
+        if tipvol == 0:
+            s50.aspirate(40, HB)
+            tipvol = 40
+        s50.dispense(20, magplate.wells(well).top())
+        tipvol -= 20
+    s50.drop_tip()
+
+    s50.delay(minutes=1)
+
+    """Bead Dispensing"""
+    # dispense 5ul of magbeads into each well
+    tipvol = 0
+    for well in magplate:
+        if tipvol == 0:
+            pick_up(s50)
+            s50.aspirate(40, magbeads)
+            tipvol = 40
+        s50.dispense(5, well)
+        tipvol -= 5
+        if tipvol == 0:
+            s50.drop_tip()
+
+    for col in mag300:
+        pick_up(m300)
+        m300.mix(5, 30, col)
+        m300.blow_out(col.top())
+        m300.drop_tip()
+
+    for _ in range(3):
+        m300.delay(minutes=20)
+        robot._driver.run_flag.wait()
+        for col in mag300:
+            pick_up(m300)
+            m300.mix(5, 30, col)
+            m300.blow_out(col.top())
+            m300.drop_tip()
+
+    """Reaction"""
+    magdeck.engage()
+    m300.delay(minutes=1)
+
+    def remove_supernatant(plate, vol):
+        m300.set_flow_rate(aspirate=50)
+        for col in plate:
+            if not m300.tip_attached:
+                pick_up(m300)
+            m300.transfer(vol, col, liqwaste, new_tip='never')
+            m300.drop_tip()
+        m300.set_flow_rate(aspirate=150)
+
+    remove_supernatant(mag300, 35)
+
+    # add PBS-T
+    def PBStransfer(vol):
+        pick_up(m300)
+        for col in mag300:
+            m300.transfer(vol, res1, col.top(), new_tip='never')
+            m300.blow_out(col.top())
+
+    # magdeck mixing
+    def magmix(time):
+        for i in range(6):
+            if i % 2 == 0:
+                magdeck.disengage()
+            else:
+                magdeck.engage()
+            m300.delay(seconds=time)
+            robot._driver.run_flag.wait()
+
+        magdeck.engage()
+        m300.delay(minutes=1)
+        robot._driver.run_flag.wait()
+
+    for t in [10, 15, 20]:
+        PBStransfer(150)
+
+        magmix(t)
+
+        if t == 20:
+            robot.pause('Add the reducing agent to the master mix. When ready \
+            to continue, click RESUME.')
+        remove_supernatant(mag300, 150)
+
+    # add 50ul of Master Mix to each well
+    tipvol = 0
+    pick_up(m300)
+    for col in mag300:
+        if tipvol == 0:
+            m300.aspirate(300, mm)
+            tipvol = 300
+        m300.dispense(50, col.top())
+        tipvol -= 50
+
+    magdeck.disengage()
+
+    # mix the wells
+    for col in mag300:
+        if not m300.tip_attached:
+            pick_up(m300)
+        m300.mix(5, 50, col)
+        m300.blow_out(col.top())
+        m300.drop_tip()
+
+    robot.pause('Please remove the plate from the Magnetic Module and place on \
+    the Temperature Module. When ready, click RESUME.')
+
+    tempdeck.set_temperature(41)
```

Comment:
move this on line 19 so that the tempdeck starts heating immediately (when the protocol starts running). Here on line 211, add the line `tempdeck.wait_for_temp()`. this ensures that the protocol will only continue if the tempdeck has heated to the target temp